#!/usr/bin/env python
PACKAGE = 'xbox_controller'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#list of parameters
gen.add('channel_0_default', int_t, 0, "default value for channel 0", default=1500, min=1000, max=2000)
gen.add('channel_0_range', int_t, 0, "range for working of channel 0", 120, 0, 200)

gen.add('channel_1_default', int_t, 0, "default value for channel 1", default=1500, min=1000, max=2000)
gen.add('channel_1_range', int_t, 0, "range for working of channel 1", 120, 0, 200)

gen.add('channel_2_max', int_t, 0, "maximum value for channel 2", 1320, 1000, 2000)
gen.add('channel_2_range', int_t, 0, "range for working of channel 2", 10, 0, 250)

gen.add('channel_3_default', int_t, 0, "default value for channel 3", default=1500, min=1000, max=2000)
gen.add('channel_3_range', int_t, 0, "range for working of channel 3", 0, 0, 100)

exit(gen.generate(PACKAGE, "xbox_controller", "Controller_init"))
